libpod/container_internal_linux.go

```diff
@@ -1399,3 +1411,42 @@ func (c *Container) getOCICgroupPath() (string, error) {
 		return "", errors.Wrapf(define.ErrInvalidArg, "invalid cgroup manager %s requested", c.runtime.config.CgroupManager)
 	}
 }
+
+func (c *Container) setupOverlayMounts() error {
+	g := generate.Generator{Config: c.config.Spec}
+	// Check if the spec file mounts contain an Overlay mount
+	// If they do, mount the overlay.
+	containerDir, err := c.runtime.store.ContainerDirectory(c.ID())
+	if err != nil {
+		return err
+	}
+
+	for i := range c.config.Spec.Mounts {
+		m := &g.Config.Mounts[i]
+		for _, o := range m.Options {
+			if o == "O" {
+				contentDir, err := overlay.TempDir(containerDir, c.RootUID(), c.RootGID())
```

Comment:
Should we make a subdirectory for these so we don't scatter them around the main directory?

Reply:
overlay.TempDir creates a the directory overlay/TMPDIR on the staticdir.  So this is already handled.